packit_service/worker/checker/koji.py

```diff
@@ -0,0 +1,65 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import logging
+
+from packit_service.worker.checker.abstract import Checker
+from packit_service.worker.events.enums import GitlabEventAction
+from packit_service.worker.events import (
+    MergeRequestGitlabEvent,
+    PullRequestGithubEvent,
+    PushGitHubEvent,
+    PushGitlabEvent,
+    PushPagureEvent,
+)
+from packit_service.constants import (
+    KOJI_PRODUCTION_BUILDS_ISSUE,
+    PERMISSIONS_ERROR_WRITE_OR_ADMIN,
+)
+from packit_service.worker.reporting import BaseCommitStatus
+from packit_service.worker.handlers.mixin import GetKojiBuildJobHelperMixin
+
+logger = logging.getLogger(__name__)
+
+
+class Permission(Checker, GetKojiBuildJobHelperMixin):
+    def pre_check(self) -> bool:
+        if (
+            self.data.event_type == MergeRequestGitlabEvent.__name__
+            and self.data.event_dict["action"] == GitlabEventAction.closed.value
+        ):
+            # Not interested in closed merge requests
+            return False
+
+        if self.data.event_type in (
+            PushGitHubEvent.__name__,
+            PushGitlabEvent.__name__,
+            PushPagureEvent.__name__,
+        ):
+            configured_branch = self.koji_build_helper.job_build_branch
+            if self.data.git_ref != configured_branch:
+                logger.info(
+                    f"Skipping build on '{self.data.git_ref}'. "
+                    f"Push configured only for '{configured_branch}'."
+                )
+                return False
+
+        if self.data.event_type == PullRequestGithubEvent.__name__:
```

Comment:
Are we missing `MergeRequestGitlabEvent`? :eyes: /o\

Reply:
I don't know, but I checked and the original code was like this one. For the moment I will leave it as it is, and if everything will work we can try to add it later. What do you think?

Reply:
Fine by me, I'll open a follow-up issue to track this and make sure we don't forget about it :)

Reply:
Created #1709